File: Program's_Contributed_By_Contributors/C++/ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
       int maxar = INT_MIN;
        int left = 0, right = height.size()-1, crar = 0;
        while (left < right){
            crar = abs(left-right) * (min(height[left], height[right]));
            if (maxar < crar) maxar = crar;

            if (height[left] < height[right]) left++;
            else right--;
        }

        return maxar; 
    }
};

File: Program's_Contributed_By_Contributors/JavaScript_Programs/container-with-most-water.js
var maxArea = function(height) {
    let left = 0;
    let right = height.length - 1;
    let maxArea = 0;
    while(left < right )
    {
        let area = Math.min(height[left],height[right]) * (right - left);
        if(area > maxArea) maxArea = area;
        if(height[left] > height[right])
        {
            right--;
        }else{
            left++;

        }
    }
    return maxArea;
};
let a = [1,8,6,2,5,4,8,3,7];
console.log(maxArea(a));